src/main/native/GCM.c

```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+					dataOffset, dataLen, outOffset, *updateOutlen);
+    		}
 #endif
+	
+	if (rc == ICC_OSSL_SUCCESS) {
+	        
+#ifdef DEBUG_GCM_DETAIL
+	   		if ( debug ) {
+    	    	gslogMessage ("DETAIL_GCM NI checking for data > 0\n");
+    		}
+#endif	
+		
+		if (dataLen > 0) {
+			
+    		
+#ifdef DEBUG_GCM_DETAIL
+	   		if ( debug ) {
+    	    	gslogMessage ("DETAIL_GCM NI ICC_AES_GCM_DecryptUpdate dataLen > 0 dataLen = %d outOffset %d updateOutlen %ld\n", 
+				dataLen, dataOffset, *updateOutlen);
+    		}
+#endif			
+			
+			rc = ICC_AES_GCM_DecryptUpdate(ockCtx, gcmCtx,NULL, 0,data + dataOffset, dataLen, out + outOffset, updateOutlen);
+			
+#ifdef DEBUG_GCM_DETAIL
+	   		if ( debug ) {
+    	    	gslogMessage ("DETAIL_GCM NI data update returned ICC_AES_GCM_DecryptUpdate returns rc = %d updateOutLen = %ld\n", rc, *updateOutlen);
+				
+    		}
+#endif		
+			
+				
 
-        if (rc == ICC_OSSL_SUCCESS) {
-
-		ret = GCM_decrypt_core (env, ockCtx, gcmCtx,
-				keyNative, keyLen,
-				ivNative, ivLen,
-				ciphertextNative, ciphertextOffset, ciphertextLen,
-				plaintextNative, plaintextOffset,
-				aadNative, aadLen, tagLen);
-	} else {
-		ret = GetPRIMITICEARRAYCRITICAL;
-	}
-
-        if (keyNative != NULL ) {
-                (*env)->ReleasePrimitiveArrayCritical(env, key, keyNative, 0);
-        }
-
-        if (ivNative != NULL ) {
-                (*env)->ReleasePrimitiveArrayCritical(env, iv,  ivNative, 0);
-        }
-
-        if (ciphertextNative != NULL ) {
-                (*env)->ReleasePrimitiveArrayCritical(env, ciphertext, ciphertextNative, 0);
-        }
-
-        if (plaintextNative != NULL ) {
-                (*env)->ReleasePrimitiveArrayCritical(env, plaintext, plaintextNative, 0);
-        }
-
-        if (aadNative != NULL ) {
-                (*env)->ReleasePrimitiveArrayCritical(env, aad, aadNative, 0);
-        }
-
-//      if ( gcmCtx != NULL) {
-//              PutGCMContext(thrGcmCtx, ockCtx);
-//      }
+			// that needs to catch a hash mismatch condition
+			if ( rc != ICC_OSSL_SUCCESS ) {
+				ockCheckStatus(ockCtx);
+				return ICC_AES_GCM_CRYPTUPDATE_FAILED;
+			}
+			else {
+				
 #ifdef DEBUG_GCM_DETAIL
-        if ( debug ) {
-                gslogMessage ("DETAIL_GCM ret=%d", (int) ret);
+	   			if ( debug ) {
+    	    		gslogMessage ("DETAIL_GCM NI data ICC_AES_GCM_DecryptUpdate Succeeded\n");
+    			}
+#endif	
+			
+			}
         }
+		else
+		{
+#ifdef DEBUG_GCM_DETAIL
+	   		if ( debug ) {
+    	    	gslogMessage ("DETAIL_GCM NI  No data to process\n");
+    		}
+#endif				
+		}
+	}
+	else {
+		/* GCM CTX Failed - no need to free it. */
+		ockCheckStatus(ockCtx);
+#ifdef DEBUG_GCM_DETAIL
+	   	if ( debug ) {
+    	    gslogMessage ("DETAIL_GCM NI  ICC_AES_GCM_DecryptUpdate ICC_AES_GCM_NEW_FAILED\n" );
+    	}
+#endif	
+		
+		return ICC_AES_GCM_CTX_NEW_FAILED;
+	}
+	return 0;
+	
+#else
```

Comment:
Will this `#else` ever be exercised since we have an `#endif` a couple of lines above?

Reply:
Not sure if this will apply to previous block. I would assume it would not.

Reply:
After further inspection it seems this pattern is matched to the statement at the top of various methods such as `#if defined(AIX) || defined(WINDOWS) || defined(MAC) || defined (LINUX) || defined(__MVS__)` this else is expected to return -1 when compiled on platforms other then this.